pkg/cmd/openshift-operators/apiserver-operator/apis/apiserver/v1/types.go

```diff
@@ -0,0 +1,48 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type OpenShiftAPIServerConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   OpenShiftAPIServerConfigSpec   `json:"spec"`
+	Status OpenShiftAPIServerConfigStatus `json:"status"`
+}
+
+type OpenShiftAPIServerConfigSpec struct {
+	// TODO I think this should eventually embed the entire master config
+	// it will end up overlaying in the following order:
+	// 1. hardcoded default
+	// 2. existing config
+	// 3. this config
+	APIServerConfig APIServerConfig `json:"apiServerConfig"`
+
+	// ImagePullSpec is the image to use
+	ImagePullSpec string `json:"imagePullSpec"`
+
+	// Version is major.minor.micro-patch?.  Usually patch is ignored.
+	Version string `json:"version"`
```

Comment:
Isn't the version embedded in the ImagePullSpec?

Reply:
> Isn't the version embedded in the ImagePullSpec?

not necessarily.  We could try to force it if we wished.